src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/ResultExecutedContext.cs

```diff
@@ -29,15 +29,31 @@ public class ResultExecutedContext : FilterContext
             IList<IFilterMetadata> filters,
             IActionResult result,
             object controller)
-            : base(actionContext, filters)
+            : this(actionContext, filters, result, controller, ResultExecutionScope.Action)
         {
-            if (result == null)
-            {
-                throw new ArgumentNullException(nameof(result));
-            }
+        }
 
-            Result = result;
+        /// <summary>
+        /// Instantiates a new <see cref="ResultExecutedContext"/> instance.
+        /// </summary>
+        /// <param name="actionContext">The <see cref="ActionContext"/>.</param>
+        /// <param name="filters">All applicable <see cref="IFilterMetadata"/> implementations.</param>
+        /// <param name="result">
+        /// The <see cref="IActionResult"/> copied from <see cref="ResultExecutingContext.Result"/>.
+        /// </param>
+        /// <param name="controller">The controller instance containing the action.</param>
+        /// <param name="resultExecutionScope">The <see cref="ResultExecutionScope"/> that the filter is currently executing in.</param>
+        public ResultExecutedContext(
+            ActionContext actionContext,
+            IList<IFilterMetadata> filters,
+            IActionResult result,
+            object controller,
+            ResultExecutionScope resultExecutionScope)
```

Comment:
I don't think we should have this concept. We could add it if we need to some day, but if you care about the scope you could just write that type of filter